### PR TITLE
Candidate validation

### PR DESCRIPTION
**Pull request by SebastianSchmidl**

### CODE REVIEW

File: src/main/scala/com/github/codelionx/distod/actors/worker/CandidateValidation.scala [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        val subClazzes = mutable.Map.empty[Int, mutable.Buffer[Int]]
+        for (tuple <- clazz) {
+          val index = leftTupleValueMapping(tuple)
+          val subClazz = subClazzes.getOrElseUpdate(index, mutable.Buffer.empty)
+          subClazz += tuple
+        }
+        subClazzes.values.map(_.toSeq).toIndexedSeq
+      }
+
+      var swap = false
```
Comment (CodeLionX) [inline src/main/scala/com/github/codelionx/distod/actors/worker/CandidateValidation.scala]: Codacy Issue found: [Avoid mutable local variables](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4581953)

File: src/main/scala/com/github/codelionx/distod/actors/worker/CandidateValidation.scala [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      var reverseSwap = false
```
Comment (CodeLionX) [inline src/main/scala/com/github/codelionx/distod/actors/worker/CandidateValidation.scala]: Codacy Issue found: [Avoid mutable local variables](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4581953)

File: src/main/scala/com/github/codelionx/distod/actors/worker/CandidateValidation.scala [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      else
+        CandidateSet.empty
+    }
+    SplitCandidateValidationResult(
+      validOds = constantOds,
+      removedCandidates = removedCandidates
+    )
+  }
+
+  def checkSwapCandidates(
```
Comment (CodeLionX) [inline src/main/scala/com/github/codelionx/distod/actors/worker/CandidateValidation.scala]: Codacy Issue found: [Method is longer than 50 lines](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4581953)

Comment (CodeLionX) [inline src/main/scala/com/github/codelionx/distod/actors/worker/CandidateValidation.scala]: Codacy Issue found: [Method is longer than 50 lines](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4581953)

File: src/main/scala/com/github/codelionx/distod/actors/worker/CandidateValidation.scala (hunk 1 above)
Comment (CodeLionX) [inline src/main/scala/com/github/codelionx/distod/actors/worker/CandidateValidation.scala]: Codacy Issue found: [Avoid mutable local variables](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4581953)

File: src/main/scala/com/github/codelionx/distod/actors/worker/CandidateValidation.scala (hunk 2 above)
Comment (CodeLionX) [inline src/main/scala/com/github/codelionx/distod/actors/worker/CandidateValidation.scala]: Codacy Issue found: [Avoid mutable local variables](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4581953)

File: src/main/scala/com/github/codelionx/distod/types/OrderDependency.scala [hunk 4]
```diff
@@ -50,7 +50,7 @@ object OrderDependency {
           val attribute1Name = names(attribute1)
           val attribute2Name = names(attribute2)
           val contextNames = context.map(names).mkString(", ")
-          s"{$contextNames}: $attribute1Name ~ $attribute2Name$orderOfSecondAttribute"
+          s"{$contextNames}: $attribute1Name↑ ~ $attribute2Name$orderOfSecondAttribute"
```
Comment (SebastianSchmidl) [inline src/main/scala/com/github/codelionx/distod/types/OrderDependency.scala:53]: Codacy Issue found: [Non ascii characters are not allowed](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4581953)

**Pull request merged**